lib/SILGen/SILGenExpr.cpp

```diff
@@ -5480,6 +5480,28 @@ RValue RValueEmitter::visitMakeTemporarilyEscapableExpr(
     visit(E->getNonescapingClosureValue()).getAsSingleValue(SGF, E);
 
   auto escapingFnTy = SGF.getLoweredType(E->getOpaqueValue()->getType());
+  auto silFnTy = escapingFnTy.castTo<SILFunctionType>();
+
+  // Handle @convention(block). No withoutActuallyEscaping verification yet.
+  if (silFnTy->getExtInfo().getRepresentation() !=
+      SILFunctionTypeRepresentation::Thick) {
+    RValue rvalue;
+    auto escapingClosure =
+        SGF.emitManagedRValueWithCleanup(SGF.B.createConvertFunction(
+            E, functionValue.ensurePlusOne(SGF, E).forward(SGF), escapingFnTy));
+    // Bind the opaque value to the escaping function.
+    SILGenFunction::OpaqueValueState opaqueValue{
+        escapingClosure,
+        /*consumable*/ true,
+        /*hasBeenConsumed*/ false,
+    };
+    SILGenFunction::OpaqueValueRAII pushOpaqueValue(SGF, E->getOpaqueValue(),
+                                                    opaqueValue);
+
+    // Emit the guarded expression.
+    rvalue = visit(E->getSubExpr(), C);
```

Comment:
Why not just return this rather than assign to rvalue.

Reply:
Will cleanup in follow-up commit.

Reply:
SGTM

Reply:
https://github.com/apple/swift/pull/15402